plugin/src/shared/sc/plugin2021/GameState.kt

```diff
@@ -139,25 +137,30 @@ class GameState @JvmOverloads constructor(
             if (orderedColors.isNotEmpty())
                 "GameState $round/$turn -> $currentColor ${if (GameRuleLogic.isFirstMove(this)) "(Start Piece: $startPiece)" else ""}"
             else "GameState finished at $round/$turn"
-
+    
+    override fun clone() = GameState(this)
+    
     override fun equals(other: Any?): Boolean {
-        return !(this === other) &&
-               other is GameState &&
-               first == other.first &&
-               second == other.second &&
-               board == other.board &&
-               turn == other.turn &&
-               currentTeam == other.currentTeam
+        return this === other ||
+               (other is GameState
+                && first == other.first
+                && second == other.second
+                && startPiece == other.startPiece
+                && board == other.board
+                && lastMove == other.lastMove
+                && lastMoveMono == other.lastMoveMono
+                && turn == other.turn)
```

Comment:
Since Kotlin doesn't have enforced semicolon to mark line endings, in ambigous cases of multi-line statements you need to put the operator at the end of the previous line (which is why `||` has to be at the end of the line).
Thus I think it's cleaner to put _every_ operator of multi-line statements at the end of the line, rather than the beginning

Reply:
though having the operators aligned improves readability imo :thinking:

Reply:
no, the `||` can also be at the beginning of the line, this isn't really an ambiguous case.

Reply:
oh, really?
I thought it couldn't know the return continues over several lines.
But maybe they just fixed that already

Reply:
If it's possible I wonder why you don't put the `||` at the beginning of the next line. That should improve readability even more, then

Reply:
because then it would be in front of the parentheses and thus not aligned either.